pandas/io/formats/style_render.py

```diff
@@ -1840,7 +1874,10 @@ def _maybe_wrap_formatter(
 
     # Replace chars if escaping
     if escape is not None:
-        func_1 = lambda x: func_0(_str_escape(x, escape=escape))
+        markupsafe = import_optional_dependency(
```

Comment:
this might be quite an expensive import operation if the formatter is applied over each cell. Will have to check this